### ISSUE TITLE
Password Change Still Broken/htpasswd Defunct?

### ISSUE THREAD

**Issue by prof79** [zevenet/zlb#22]
Hello,

I'm using a ZEVENET VM installed from the ZEVENET 4.01 Community ISO image.

I've noticed issue #18 and made a symbolic link `/usr/local/zlb` -> `/usr/local/zenloadbalancer` which should be fine as well I suppose. However, when resetting the password from the web GUI the first line of `/usr/local/zenloadbalancer/app/cherokee/etc/cherokee/.htpasswd` goes to "`admin:`" (without a hash after the colon).

What makes me wonder is that it seems like the `htpasswd` binary is broken in the distribution.

When trying to run `/usr/local/zenloadbalancer/app/cherokee/sbin/htpasswd` as root from the command line I get an error:

`-bash: /usr/local/zenloadbalancer/app/cherokee/sbin/htpasswd: cannot execute binary file: Exec format error`

Also getting this on a second VM.

Thanks!

**Comment by prof79** [comment]
After some research/experimentation I've new findings:

* `htpasswd` seems to be irrelevant to the web GUI password change yet it would be helpful to have a non-broken version handy.
* Password change does indeed work but is error-prone regarding special characters (see below).
* Internet Explorer shows about 15+ authentication dialogs on password change which is confusing/misleading. (IE 11 x86 on an up-to-date Windows 8.1 Pro x64)

Regarding the blanked `admin:` line, reproduce as follows:

* Start from a ZLB with password set to `admin`.
* Change the password to `-` or `.` or something containing these characters.
* This will/should work. (the auth on the out/out.cgi is a nuisance btw)
* Log-in again using the new password.
* Try changing the password back to something simple like `admin`.
* Now the `admin` line in the `.htpasswd` file will be invalid.

To sum it up it seems like the contents of the "`Current password:`" box are not properly encoded/sanitized thus invalidating `.htpasswd` on special character input.

**Comment by emiliocampos-zevenet** [comment]
Dear prof79, right, it seems like value "." or "-" is accepted in gui but is not parsed by the login process after all. 

Issue reproduced.

**Comment by thiesuh** [comment]
Yesterday i updatet zevenet v4.0.3 to v4.0.4. After the update the password resets automatically to the default password en i had to change it back. Would like to see that it keeps the password we changed it to.

**Comment by nevola** [comment]
Hi! Could you please detail the steps to reproduce it? How did you
installed zevenet and how have you tried to update it?

Thanks!

**Comment by GaderyPoluki** [comment]
The problem still exists in v4.0.4. Enterprise Edition. I reset my password today using @ and $ and both of the characters were ignored. Thanks to this article at least I was able to come up with the reason why the new password doesn't work. The web interface gave a response that the password was changed but the system actually ignored these characters.

**Comment by emiliocampos-zevenet** [comment]
Did you try 4.0.5? I remember we worked in this matter in past.

**Comment by GaderyPoluki** [comment]
Well no, I didnt try but when the root password was improperly updated on the production environment an upgrade was the last thing I would think of. I had to take the new password and remove the special characters one by one and then in various combinations because since were accepted and some not. So I actually had to guess my password :-(.

**Comment by emiliocampos-zevenet** [comment]
Thanks for the information. We are currently working on new zevenet 5 version, this issue will be solved after all. 

Thanks!
